### DISCUSSION TITLE
Modelleerprincipes - P3.  Alleen directe eigenschappen

### DISCUSSION THREAD

**Discussion by pmaria** [Geonovum/disgeo-imsor#62] [General]
Modelleerprincipe [[P3]](https://geonovum.github.io/disgeo-imsor/modelleerprincipes/#p3-alleen-directe-eigenschappen) - Alleen directe eigenschappen komen voor op een objecttype.

Discussie uit expertgroepsessie:

- Pano: als je dit principe niet toepast moet je elke eigenschap uitpluizen om te bepalen waar het over gaat (wanneer je informatie wil gebruiken in een eigen context). Dan kun je het ook niet gemakkelijk gebruiken op een geautomatiseerde manier. Wel is een van de implicaties van dit principe dat het leidt tot een genormaliseerd informatiemodel.
- Lennart: dus wat je bedoelt is dat je bv bij BAG pand en BGT pand, omdat ze over hetzelfde ding gaan, de eigenschappen van die twee kan optellen bij elkaar. Maar in MIM houden we daarom het conceptueel en logisch model gescheiden. In een conceptueel model zou je dit uit elkaar moeten kunnen houden [intrinsieke vs andere eigenschappen], maar in een logisch model wil je misschien dingen samenschuiven omdat het handiger is in de implementatie. Ik ben het eens met de stelling en rationale, de uitleg is mooi, alleen zullen er zeker mensen zijn die niet blij worden van deze uitwerking. 
- Lennart: een alternatieve benadering zou zijn om van elk attribuut aan te geven of het een echte eigenschap is van het object dat je ermee beschrijft. 
- Rik: zijn de IDs ook hetzelfde? 
- Pano: voor nu niet, het idee is dat we dit later wel realiseren.
- Niels: dit raakt wel aan de kern van hoe we het met z'n allen voor ons zien [de samenhang] - is de SOR een registratie, een laag over andere registraties? Ik had eigenlijk verwacht dat in de Architectuurbeschrijving hier al meer richting aan zou worden gegeven. 
- Rik: en dan zie ik veel verbanden met de "verwantschapslaags" uit het UOI project.
- Pano: let wel op dat het UOI idee fundamenteel anders is - met zo'n laag geef je relaties weer tussen informatieobjecten, SOR probeert een ID toe te kennen aan objecten in de werkelijkheid. 
- Niels: in de nauwere context was ik er van uitgegaan dat er een beweging zou komen om de huidige registraties samen te laten komen in één registratie. We hebben een overgangsstrategie hiervoor nodig, maar hier wordt ook een scenario geschetst waarbij de registraties kunnen blijven bestaan (waarop een verwantschapslaag kan worden gerealiseerd).
- Lennart: volgens mij is het heel duidelijk dat het niet de bedoeling is dat het één registratie wordt.
- Dick: ik schat in dat het stapsgewijs gaat gebeuren, zodat we uiteindelijk [ver in de toekomst] wel een SOR hebben en geen onderliggende registraties. Maar op beleidsniveau moet het nog helder worden hoe dit zich gaat ontwikkelen. 
- Lennart: als het in de werkelijkheid om hetzelfde ding gaat en mensen dit willen weten, lijkt me het een goed principe om het op deze manier te modelleren - en daarvoor is het nodig dat je weet wat bij het object hoort (intrinsiek) en wat niet. Je heb verschillende perspectieven waarmee je naar het object kijkt, het lijkt me moeilijk om dit in elkaar te schuiven.
- Linda: dus iedereen is het eens met dit principe? Lennart en Niels beamen dit. Niemand zegt het er niet mee eens te zijn. 
- Lennart: alleen wat betreft de implicaties zou ik nog 1 of 2 alternatieven verzinnen.

**Comment by lennartvanbergen** [comment]
Voorbeeld: Document zit niet in de BAG als objecttype (zie BAG wet). Deze uitwerking kan mijn inziens niet.

**Comment by lennartvanbergen** [comment]
Op logisch niveau mag je dingen toevoegen bij een objecttype als dat handig is voor de implementatie. Op conceptueel IM niveau hoort document mijn inziens inderdaad niet. Maar logisch mag het gewoon. 

SOR is logisch. 

Document is in de BAG geen object (zie BAG wet), het is een relatie naar een Document, en het Document zit buiten de BAG (bv. via MIM: externe koppeling). Dit valt dus niet te modelleren als Document. Het is een relatie naar een Document. Maar op logisch niveau mag je dit toevoegen aan Pand. 

Ik denk dat we toe moeten willen naar: 
- op conceptueel IM niveau zijn alle eigenschappen directe eigenschappen 
- omdat het mag op logisch niveau, mag je document toevoegen bij Pand, en dan wil je denk ik metadata kunnen toevoegen op logisch IM niveau om aan te duiden of het een directe eigenschap is. 

Kortom, een aanduiding/metadata of iets een directe eigenschap is, is nodig op logisch niveau. Op logisch niveau heb je immers meer vrijheid en mag e.e.a. en kunnen we het op logisch niveau niet zeggen: dit moet je altijd oplossen met zuiver modelleren.
